projects/plugins/boost/app/modules/cache/cache.php

```diff
@@ -0,0 +1,168 @@
+	 *
+	 * If a request is in the backend, or is a POST request, or is not an
+	 * html request, it is not cacheable.
+	 * The filter boost_cache_cacheable can be used to override this.
+	 *
+	 * @return bool
+	 */
+	public function is_cacheable() {
+		if ( ! apply_filters( 'boost_cache_cacheable', $this->request_uri ) ) {
+			return false;
+		}
+
+		if ( $this->is_backend() ) {
+			return false;
+		}
+
+		if ( ( isset( $_SERVER['REQUEST_METHOD'] ) && $_SERVER['REQUEST_METHOD'] === 'POST' ) ) {
+			return false;
+		}
+
+		if ( isset( $_SERVER['HTTP_ACCEPT'] ) && strpos( $_SERVER['HTTP_ACCEPT'], 'text/html' ) === false ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			return false;
+		}
+
+		return true;
+	}
+
+	/*
+	 * Returns a key to identify the visitor's cache file.
+	 * It is based on the REQUEST_URI, the cookies and the page GET parameters.
+	 *
+	 * @param string $request_uri (optional) The request uri to use to calculate the cache key. Defaults to the current request uri.
+	 *
+	 * @return string
+	 */
+	private function calculate_cache_key( $request_uri = '' ) {
+		if ( $request_uri === '' ) {
+			$request_uri = $this->request_uri;
+		}
+		$cookies = isset( $_COOKIE ) ? $_COOKIE : array();
+		$get     = isset( $_GET ) ? $_GET : array(); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		$key_components = apply_filters(
+			'boost_cache_key_components',
+			array(
+				'cookies'     => $cookies,
+				'request_uri' => $request_uri,
+				'get'         => $get,
+			)
+		);
+
+		return md5( json_encode( $key_components ) ); // phpcs:ignore WordPress.WP.AlternativeFunctions.json_encode_json_encode
+	}
+
+	/*
+	 * Returns a key to identify the visitor's cache file.
+	 * It uses calculate_cache_key() to calculate the key.
+	 * Without a parameter, it will use the current request uri, and cache that
+	 * value in $this->cache_key.
+	 *
+	 * @param string $request_uri (optional) The request uri to use to calculate the cache key. Defaults to the current request uri.
+	 * @return string
+	 */
+	public function cache_key( $request_uri = '' ) {
+		if ( $request_uri !== '' ) {
+			return $this->calculate_cache_key( $request_uri );
+		}
+
+		if ( ! $this->cache_key ) {
+			$this->cache_key = $this->calculate_cache_key();
+		}
+		return $this->cache_key;
+	}
+
+	/*
+	 * Returns a key to identify the path to the visitor's cache file.
+	 * Without a parameter it uses the current request uri, and caches that in
+	 * $this->path_key.
+	 *
+	 * @param string $request_uri (optional) The request uri to use to calculate the path key. Defaults to the current request uri.
+	 * @return string
+	 */
+	public function path_key( $request_uri = '' ) {
+		if ( $request_uri !== '' ) {
+			return md5( $request_uri );
+		}
+
+		if ( ! $this->path_key ) {
+			$this->path_key = md5( $this->request_uri );
+		}
+		return $this->path_key;
+	}
```

Comment:
What is the difference between a `path_key` and a `cache_key`? I can't see this function used anywhere, so I don't understand its purpose.

Reply:
The path_key here is the md5 of the request_uri, as you can see. I forgot to use it in the file cache in calculate_path(), but I've just fixed that. I removed path_key, but then realised it should be used and put it back again.

We could have multiple cache files for a particular URL. One user could be completely anonymous, another may leave a comment, another will click a cookie banner, all resulting in different cookies.
With the path and cookie keys split up, it's simple to delete all the cache files for that URL in one go, if a new comment is made or the page is edited.